src/test/coins_tests.cpp

```diff
@@ -145,8 +145,8 @@ BOOST_AUTO_TEST_CASE(coins_cache_simulation_test)
             // Determine whether to test HaveCoin before or after Access* (or both). As these functions
             // can influence each other's behaviour by pulling things into the cache, all combinations
             // are tested.
-            bool test_havecoin_before = (insecure_rand() & 0x3) == 0; // TODO change to InsecureRandBits(2) when backporting Bitcoin #10321
-            bool test_havecoin_after = (insecure_rand() & 0x3) == 0; // TODO change to InsecureRandBits(2) when backporting Bitcoin #10321
+            bool test_havecoin_before = (InsecureRandBits(2)) == 0;
```

Comment:
Why the unnecessary ()?

Reply:
https://github.com/dashpay/dash/pull/3021/commits/8c6f51ee19464d62e282246a8e3195aaefd06d39

Reply:
https://github.com/dashpay/dash/pull/3021/commits/84b288ea7a3a00d9f4928a3ce29af8fef32f2117